src/firefly/js/ui/JoyToMUIThemeWrapper.jsx

```diff
@@ -0,0 +1,57 @@
+import React, {useEffect} from 'react';
+import {THEME_ID as MATERIAL_THEME_ID,
+    experimental_extendTheme as materialExtendTheme,
+    Experimental_CssVarsProvider as MaterialCssVarsProvider,
+    useColorScheme as useMaterialColorScheme
+} from '@mui/material/styles';
+import {CssVarsProvider as JoyCssVarsProvider} from '@mui/joy/styles';
+import {useColorMode, getTheme} from 'firefly/ui/FireflyRoot';
+
+
+/**
+ * This is needed to use MUI X components in JoyUI, otherwise they will throw errors because they expect MUI theme.
+ * Adapted from the example code present at https://mui.com/x/react-date-pickers/custom-field/#using-another-input
+ * and https://mui.com/joy-ui/integrations/material-ui/
+ *
+ * @param p
+ * @param p.children
+ */
+export function JoyToMUIThemeWrapper({children}) {
+    const {activeMode: joyMode} = useColorMode();
+    const joyTheme = getTheme();
+
+    // To sync MUI's mode with JoyUI's mode, when JoyUI's mode changes
+    const SyncThemeMode = ({ mode }) => {
+        const { setMode: setMaterialMode } = useMaterialColorScheme();
+        useEffect(() => {
+            setMaterialMode(mode);
+        }, [mode, setMaterialMode]);
+        return null;
+    };
+
+    // Override the default MUI appearance of wrapped component (esp font, color, etc.) with JoyUI theme
+    // Note: MUI and JoyUI theme objects are not identical by design, so following doesn't cover all cases
+    const materialTheme = materialExtendTheme({
+        typography: {
+            fontFamily: joyTheme.fontFamily.body
+        },
+        colorSchemes: Object.fromEntries(['light', 'dark'].map((k)=> [k, {
+            palette: {
+                // joyUI primary colors remain same for both dark and light mode
+                primary: {
+                    main: joyTheme.palette.primary[500],
+                    dark: joyTheme.palette.primary[600], //used by MUI in hover
+                }
+            }
+        }]))
+    });
+
+    return (
+        <MaterialCssVarsProvider theme={{ [MATERIAL_THEME_ID]: materialTheme }}>
+            <JoyCssVarsProvider>
+                <SyncThemeMode mode={joyMode} />
+                {children}
+            </JoyCssVarsProvider>
+        </MaterialCssVarsProvider>
+    );
```

Comment:
This whole thing looks very messy. I am sure you are doing correctly but I am guessing as JoyUI becomes more mature that will be a better way. We should be watching for that and upgrade when it is available.

Reply:
Yes they have also mentioned that in the documentation I linked in docstring.